src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -662,9 +704,20 @@ private bool StartTag(Stack<Tuple<HtmlSymbol, SourceLocation>> tags, IDisposable
                 tagName = new HtmlSymbol(CurrentLocation, String.Empty, HtmlSymbolType.Unknown);
             }
 
+            // Need to check if we need to back-track.
+            if (bangSymbol != null)
+            {
+                // Need to ensure the Tokenizer is pointing to the correct location.
+                PutCurrentBack();
+                PutBack(bangSymbol);
+                NextToken();
```

Comment:
Ahhhh looks like this change is still sitting on my local box, I transitioned this bit to use the `Lookahead` helper. Will update.